shared/api/config.go

```diff
@@ -0,0 +1,42 @@
+package api
+
+type SystemConfig struct {
+	Network  ConfigNetwork  `json:"network"  yaml:"network"`
+	Security ConfigSecurity `json:"security" yaml:"security"`
+
+	Group string `json:"group" yaml:"group"` // Group name the local unix socket should be chown'ed to
+}
+
+type ConfigNetwork struct {
+	Address        string `json:"ip,omitempty"              yaml:"ip,omitempty"`
+	Port           int    `json:"port,omitempty"            yaml:"port,omitempty"`
+	WorkerEndpoint string `json:"worker_endpoint,omitempty" yaml:"worker_endpoint,omitempty"`
+}
+
+type ConfigSecurity struct {
+	// An array of SHA256 certificate fingerprints that belong to trusted TLS clients.
+	TrustedTLSClientCertFingerprints []string `json:"trusted_client_fingerprints" yaml:"trusted_client_fingerprints"`
+
+	OIDC    ConfigOIDC    `json:"oidc"    yaml:"oidc"`
+	OpenFGA ConfigOpenFGA `json:"openfga" yaml:"openfga"`
+}
+
+type ConfigOIDC struct {
+	Issuer   string `json:"issuer,omitempty"    yaml:"issuer,omitempty"`
+	ClientID string `json:"client.id,omitempty" yaml:"client.id,omitempty"`
+	Scope    string `json:"scopes,omitempty"    yaml:"scopes,omitempty"`
+	Audience string `json:"audience,omitempty"  yaml:"audience,omitempty"`
+	Claim    string `json:"claim,omitempty"     yaml:"claim,omitempty"`
+}
+
+type ConfigOpenFGA struct {
+	APIToken string `json:"api.token,omitempty" yaml:"api.token,omitempty"`
+	APIURL   string `json:"api.url,omitempty"   yaml:"api.url,omitempty"`
+	StoreID  string `json:"store.id,omitempty"  yaml:"store.id,omitempty"`
+}
```

Comment:
I previously had the settings with a dot (`.`) as well, but now I felt, it might be clearer to have it with underscore and since the format of the config file changed in breaking ways anyway, I decided to change this as well in https://github.com/FuturFusion/operations-center/pull/208. I don't have a strong opinion on this, but I see some value in having it consistent between Migration Manager and Operations Center. What do you guys think?

Reply:
Having dots in json keys definitely feels a bit odd to me, we should do underscores :)